### PR TITLE
Unit test attack handler

### PR DESCRIPTION
**Pull request by saeidalz13**
# Overview

Introduces the full unit testing for attack logic that was absent before. Unit tests now ends the game similar to a real-world scenario.
This will be also useful in case of rematch testing

### CODE REVIEW

**Comment by saeidalz13** [comment]
As soon you merge this, I'll update `rematch` branch and push it. After that I'll write unit tests for `rematch`

File: api/sessions_api.go [hunk 1]
```diff
@@ -157,8 +157,6 @@ sessionLoop:
 				}
 			}
 
-			// attacker turn is set to false
-			resp.Payload.IsTurn = false
```
Comment (mori-ahk) [inline api/sessions_api.go:161]: is this intentional?

**Review by mori-ahk — APPROVED** [review]
just a comment, the rest LGTM

File: api/sessions_api.go (hunk 1 above)
Comment (saeidalz13) [reply api/sessions_api.go:161]: So bizarre that i don't see it on my end!

**Pull request merged**